sdcm/utils/nemesis_utils/indexes.py

```diff
@@ -125,10 +128,91 @@ def drop_index(session, ks, index_name) -> None:
     session.execute(SimpleStatement(f'DROP INDEX {ks}.{index_name}'), timeout=300)
 
 
-def drop_materialized_view(session, ks, view_name) -> None:
+def drop_materialized_view(session, ks, view_name, timeout=300) -> None:
     LOGGER.info('start dropping MV: %s.%s', ks, view_name)
     with EventsFilter(
             event_class=DatabaseLogEvent.DATABASE_ERROR,
             regex=".*Error applying view update.*",
             extra_time_to_expiration=180):
-        session.execute(SimpleStatement(f'DROP MATERIALIZED VIEW {ks}.{view_name}'), timeout=300)
+        session.execute(SimpleStatement(f'DROP MATERIALIZED VIEW {ks}.{view_name}'), timeout=timeout)
+
+
+def create_materialized_view(session, ks_name, base_table_name, mv_name, mv_partition_key, mv_clustering_key,  # noqa: PLR0913
+                             mv_columns='*', speculative_retry=None, read_repair=None, compression=None,
+                             gc_grace=None, compact_storage=False):
+
+    # Fix quotes for column names, only use quotes where needed
+    if mv_columns != '*':
+        mv_columns = [
+            cql_quote_if_needed(cql_unquote_if_needed(col))
+            for col in (mv_columns if isinstance(mv_columns, list) else list(mv_columns))
+        ]
+    mv_partition_key = [
+        cql_quote_if_needed(cql_unquote_if_needed(pk))
+        for pk in (mv_partition_key if isinstance(mv_partition_key, list) else list(mv_partition_key))
+    ]
+    mv_clustering_key = [
+        cql_quote_if_needed(cql_unquote_if_needed(cl))
+        for cl in (mv_clustering_key if isinstance(mv_clustering_key, list) else list(mv_clustering_key))
+    ]
+
+    where_clause = ' and '.join([f'{kc} is not null' for kc in mv_partition_key + mv_clustering_key])
+    select_clause = ', '.join(mv_columns)
+    pk_clause = ', '.join(mv_partition_key)
+    cl_clause = ', '.join(mv_clustering_key)
+
+    query = f"CREATE MATERIALIZED VIEW {ks_name}.{mv_name} AS SELECT {select_clause} FROM {ks_name}.{base_table_name} " \
+        f"WHERE {where_clause} PRIMARY KEY ({pk_clause}, {cl_clause}) WITH comment='test MV'"
+    if compression is not None:
+        query += f" AND compression = {{ 'sstable_compression': '{compression}Compressor' }}"
+    if read_repair is not None:
+        query += f" AND read_repair_chance={read_repair}"
+    if gc_grace is not None:
+        query += f" AND gc_grace_seconds={gc_grace}"
+    if speculative_retry is not None:
+        query += f" AND speculative_retry='{speculative_retry}'"
+    if compact_storage:
+        query += ' AND COMPACT STORAGE'
+
+    LOGGER.debug(f'MV create statement: {query}')
+    session.execute(query, timeout=600)
+
+
+def create_mv_for_table(session, keyspace_name, base_table_name, view_name):
```

Comment:
`sdcm.tester.ClusterTester.create_materialized_view` how about this? can it be removed now?

Reply:
removed